datafusion/physical-expr-common/src/sort_expr.rs

```diff
@@ -409,6 +405,36 @@ impl LexOrdering {
     pub fn truncate(&mut self, len: usize) {
         self.inner.truncate(len)
     }
+
+    /// Merge the contents of `other` into `self`, removing duplicates.
+    pub fn merge(mut self, other: LexOrdering) -> Self {
+        self.inner = self.inner.into_iter().chain(other).unique().collect();
+        self
+    }
+
+    /// Converts a `LexRequirement` into a `LexOrdering`.
```

Comment:
I also included this change in https://github.com/apache/datafusion/pull/13222 (mostly as a note that this will be a logical conflict)